File: Sources/CountriesFlagsObjC/source/Countries/UIImage+AlandIslands.m
//
//  UIImage+AlandIslands.m
//  CountriesFlags
//

#import "Countries/UIImage+AlandIslands.h"

@implementation UIImage (AlandIslands)

+ (UIImage * _Nullable)flagAlandIslandsWithSize:(CGSize)size {
    UIColor* colorBlue = [UIColor colorWithRed:0.0f
                                         green:100.0f/255.0f
                                          blue:174.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorOrange = [UIColor colorWithRed:1.0f
                                           green:211.0f/255.0f
                                            blue:0.0f
                                           alpha:1.0f];
    UIColor* colorRed = [UIColor colorWithRed:219.0f/255.0f
                                        green:15.0f/255.0f
                                         blue:22.0f/255.0f
                                        alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorOrange setFill];
    // horizontal line
    UIRectFill(CGRectMake(0.0f, size.height / 837.0f * 295.0f, size.width, size.height / 837.0f * 247.0f));
    // vertical line
    UIRectFill(CGRectMake(size.width / 1280.0f * 394.0f, 0.0f, size.width / 1280.0f * 246.0f, size.height));
    
    [colorRed setFill];
    // horizontal line
    UIRectFill(CGRectMake(0.0f, size.height / 837.0f * 369.0f, size.width, size.height / 837.0f * 99.0f));
    // vertical line
    UIRectFill(CGRectMake(size.width / 1280.0f * 468.0f, 0.0f, size.width / 1280.0f * 99.0f, size.height));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+AlandIslands.swift
//
//  UIImage+AlandIslands.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Åland "wikipedia"
     
     Generates the flag of Aland islands.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Aland islands.
     */
    internal static func flagAlandIslands(with size: CGSize) -> UIImage? {
        let colorBlue = UIColor(red: 0.0,
                                green: 100.0 / 255.0,
                                blue: 174.0 / 255.0,
                                alpha: 1.0)
        let colorYellow = UIColor(red: 1.0,
                                  green: 211.0 / 255.0,
                                  blue: 0.0,
                                  alpha: 1.0)
        let colorRed = UIColor(red: 219.0 / 255.0,
                               green: 15.0 / 255.0,
                               blue: 22.0 / 255.0,
                               alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorBlue.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorYellow.setFill()
        // horizontal line
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 837.0 * 295.0),
                          size: CGSize(width: size.width, height: size.height / 837.0 * 247.0)))
        // vertical line
        UIRectFill(CGRect(origin: CGPoint(x: size.width / 1280.0 * 394.0, y: 0.0),
                          size: CGSize(width: size.width / 1280.0 * 246.0, height: size.height)))

        colorRed.setFill()
        // horizontal line
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 837.0 * 369.0),
                          size: CGSize(width: size.width, height: size.height / 837.0 * 99.0)))
        // vertical line
        UIRectFill(CGRect(origin: CGPoint(x: size.width / 1280.0 * 468.0, y: 0.0),
                          size: CGSize(width: size.width / 1280.0 * 99.0, height: size.height)))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
